curator-examples/src/main/java/discovery/DiscoveryExample.java

```diff
@@ -192,6 +194,14 @@ private static void listInstances(ServiceDiscovery<InstanceDetails> serviceDisco
                 }
             }
         }
+        catch ( Exception e )
+        {
+            if ( e instanceof KeeperException.NoNodeException )
```

Comment:
which instruction can throw this exception ? is it possible that it is throws inside the inner loop ? 
in that case we could end the loop prematurely .
that do you think ?

Reply:
When there is no one service in the registration call list will directly exit the process, and in fact should be prompted without any services, unless it is other exceptions, do you think?

Reply:
Maybe, if it's not a NoNodeException, I should throw it to stop the execution of the program, right?

Reply:
let's make it simple.
if you found this problem with NoNode then let's fix only it.

if you have ever saw other errors, then it is worth to do more

what do you think ?

Reply:
OK, it has been modified